src/main.cpp

```diff
@@ -4065,7 +4071,16 @@ bool CheckBlock(const CBlock& block, CValidationState& state, bool fCheckPOW, bo
 
     if (block.IsProofOfStake()) {
         // Coinbase output should be empty if proof-of-stake block
-        if (block.vtx[0].vout.size() != 1 || !block.vtx[0].vout[0].IsEmpty())
+        int commitpos = GetWitnessCommitmentIndex(block);
+        if (! hasEnabledStakingOnSegWit()) {
```

Comment:
remove space after `!` and just use `IsSporkActive(SPORK_19_STAKING_ON_SEGWIT)`. Also possibly needs to be moved into ContextualCheckBlock?